WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomerSection/CustomerSearchUICommand.swift

```diff
@@ -0,0 +1,80 @@
+import Foundation
+import Yosemite
+
+/// Implementation of `SearchUICommand` for Customer search.
+///
+final class CustomerSearchUICommand: SearchUICommand {
+
+    typealias Model = Customer
+    typealias CellViewModel = TitleAndSubtitleAndStatusTableViewCell.ViewModel
+    typealias ResultsControllerModel = StorageCustomer
+
+    var searchBarPlaceholder: String = Localization.searchBarPlaceHolder
+
+    var searchBarAccessibilityIdentifier: String = "customer-search-screen-search-field"
+
+    var cancelButtonAccessibilityIdentifier: String = "customer-search-screen-cancel-button"
+
+    var resynchronizeModels: (() -> Void) = {}
+
+    var onDidSelectSearchResult: ((Customer) -> Void)
+
+    private let siteID: Int64
+
+    init(siteID: Int64, onDidSelectSearchResult: @escaping ((Customer) -> Void)) {
+        self.siteID = siteID
+        self.onDidSelectSearchResult = onDidSelectSearchResult
+    }
+
+    func createResultsController() -> ResultsController<StorageCustomer> {
+        let storageManager = ServiceLocator.storageManager
+        let descriptor = NSSortDescriptor(keyPath: \StorageCustomer.customerID, ascending: false)
+        return ResultsController<StorageCustomer>(storageManager: storageManager, sortedBy: [descriptor])
+    }
+
+    func createStarterViewController() -> UIViewController? {
+        nil
+    }
+
+    func createCellViewModel(model: Customer) -> TitleAndSubtitleAndStatusTableViewCell.ViewModel {
+        return CellViewModel(
+            id: "\(model.customerID)",
+            title: "\(model.firstName ?? "") \(model.lastName ?? "")",
```

Comment:
This... might be the right thing to do given our data source, but names are complicated, and simply concatenating them isn't correct in all locales.

[See `PersonNameComponentsFormatter` docs](https://developer.apple.com/documentation/foundation/personnamecomponentsformatter#) for more info. There's not _necessarily_ anything to do here (as I don't know whether we have enough in our data source to correctly make a formatter) but I thought it would be useful for you to know about if you don't already.